hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -0,0 +1,297 @@
+                        # todo: use config instead of 8.0
+                        due_hours -= leave.compute_work_day(
+                            att_day.date) * 8
+
+            att_day.due_hours = due_hours
+
+    @api.multi
+    @api.depends('attendance_ids.worked_hours')
+    def _compute_worked_hours(self):
+        for att_day in self.filtered('attendance_ids'):
+            total_hours = 0
+            for attendance in att_day.attendance_ids:
+                total_hours += attendance.worked_hours
+            att_day.worked_hours = total_hours
+
+    @api.multi
+    @api.depends('worked_hours')
+    def _compute_logged_hours(self):
+        # todo: at verify if it's still useful
+        pass
+
+    # @api.multi
+    # @api.depends('break_ids')
+    # def _compute_due_break(self):
+    #     for att_day in self:
+    #         more_hours = att_day.worked_hours > att_day.due_hours
+    #         hours = att_day.worked_hours if more_hours else att_day.due_hours
+    #         # todo: verify
+    #         rule = self.env['hr.attendance.rules'].search([
+    #             ('threshold', '>', hours),
+    #         ])
+    #         if rule:
+    #             att_day.due_break_min = rule[0].due_break
+    #             att_day.due_break_total = rule[0].due_break_total
+    #         else:
+    #             att_day.due_break_min = 0
+    #             att_day.due_break_total = 0
+
+    @api.multi
+    @api.depends('break_ids')
+    def _compute_break_max(self):
+        for att_day in self.filtered('break_ids'):
+            att_day.break_max = max(att_day.break_ids.mapped('duration'))
+
+    @api.multi
+    @api.depends('break_ids')
+    def _compute_break_total(self):
+        for att_day in self.filtered('break_ids'):
+            att_day.break_max = sum(att_day.break_ids.mapped('duration'))
+
+    @api.multi
+    @api.depends('worked_hours', 'due_hours')
+    def _compute_extra_hours(self):
+        for att_day in self.filtered('worked_hours'):
+            att_day.extra_hours = att_day.worked_hours - att_day.due_hours
+
+    @api.multi
+    def _compute_weighting(self):
+        return 1
+
+    ##########################################################################
+    #                             PUBLIC METHODS                             #
+    ##########################################################################
+    @api.model
+    def create(self, vals):
+        rd = super(HrDayAttendance, self).create(vals)
+
+        cal_att_date = fields.Date.from_string(rd.date)
+
+        # link to schedule (resource.calendar.attendance)
+        att_day = cal_att_date.weekday()
+        cal_att_ids = rd.employee_id.contract_id.working_hours.attendance_ids
+        current_cal_att = cal_att_ids.filtered(
+            lambda a: int(a.dayofweek) == att_day)
+
+        for cal_att_id in current_cal_att:
+            start = cal_att_id.date_from
+            start = start if start else date.min
```

Comment:
You can simply merge this in the previous line by doing: `start = cal_att_id.date_from or date.min`